src/audio/host-zephyr.c

```diff
@@ -213,6 +215,19 @@ static int host_copy_one_shot(struct host_data *hd, struct comp_dev *dev, copy_c
 		return ret;
 	}
 
+	if (IS_ENABLED(CONFIG_DMA_NXP_SOF_HOST_DMA)) {
+		/* SRC/DEST addresses have changed so the DMAC needs
+		 * to be re-configured.
+		 */
+		blk.source_address = local_elem->src;
+		blk.dest_address = local_elem->dest;
+		blk.block_size = local_elem->size;
+		/* head block currently points to a stack address so use new
+		 * block for the re-configuration to avoid any "surprises".
+		 */
+		hd->z_config.head_block = &blk;
```

Comment:
sorry, what exactly does your comment above mean? Indeed this code stores a stack address into a persistent object and that indeed isn't a good idea and should be fixed, so which new block is your comment referring to, that fixes any surprises?

Reply:
Ah, what I wanted to say with that comment is that on each `host_copy_one_shot()` you use a new block placed on the stack (i.e: the `blk` variable).

Reply:
Fixed. Hopefully it's a bit better now?

Reply:
@LaurentiuM1234 but what my comment actually was around, is that you shouldn't store (at least not permanently) a pointer to an object on stack (`blk`) in a permanent object (`hd->z_config`). So that assignment isn't a very good idea. You can make it kind of acceptable if you reset the pointer to NULL again *inside* the same function before leaving it like
```
if (IS_ENABLED(CONFIG_DMA_NXP_SOF_HOST_DMA)) {
    ...
    hd->z_config.head_block = &blk;
    ...
}
...
ret = dma_config(hd->chan->dma->z_dev, hd->chan->index, &hd->z_config);
/* *** clear the pointer to a stack object! *** */
hd->z_config.head_block = NULL;
```
assuming that your `dma_config()` doesn't "steal" that pointer to store it in a different persistent object, but that would be a bug of its own then.

Reply:
ACK. Then I think I'm going to go for allocating the block configuration on the heap just like we do on `dai-zephyr`.

Reply:
dcd2bd31bf0de9cd698f3342440d8151abcaa1a8 should address this now

Reply:
if that object has to be persistent, then yes, it has to be allocated on heap. But if it's only used by the function and from other functions, used from it, i.e. while the stack frame is valid, then you can keep it on stack and then reset the pointer to `NULL` at the end of the function.